src/scripts/index.js

```diff
@@ -0,0 +1,336 @@
+const name = profile.querySelector(".profile__name");
+const about = profile.querySelector(".profile__about");
+const editBtn = profile.querySelector("#profile-btn");
+const avatarBtn = profile.querySelector("#avatar-btn");
+
+function setData(data) {
+  setInfo(data);
+  setAvatar(data);
+}
+
+function setInfo(data) {
+  name.textContent = data.name;
+  about.textContent = data.about;
+}
+
+function setAvatar(data) {
+  avatar.src = data.avatar;
+}
+
+function onInfoEdit() {
+  setInputValues(profileForm, profileInputList, {
+    name: name.textContent,
+    about: about.textContent,
+  });
+  openModal(profileModal);
+}
+
+function onAvatarEdit() {
+  openModal(avatarModal);
+}
+
+editBtn.addEventListener("click", onInfoEdit);
+avatarBtn.addEventListener("click", onAvatarEdit);
+
+api.getUser().then(setData);
+
+// Form
+function setInputValues(form, inputList, inputValues) {
+  if (inputValues) {
+    inputList.forEach((input) => {
+      if (inputValues[input.id]) {
+        input.value = inputValues[input.id];
+      } else {
+        input.reset();
+      }
+    });
+  } else {
+    form.reset();
+  }
+
+  checkFormValidity(form, settings);
+}
+
+function getInputValues(inputList) {
+  const inputValues = {};
+  inputList.forEach((input) => {
+    inputValues[input.id] = input.value;
+  });
+
+  return inputValues;
+}
+
+const savingText = "Saving...";
+const deletingText = "Deleting...";
+function setLoadingText(button, text) {
+  const oldText = button.textContent;
+  return function (isSubmiting) {
+    if (isSubmiting) {
+      button.textContent = text;
+    } else {
+      button.textContent = oldText;
+    }
+  };
+}
+
+// Modal
+let currentModal;
+
+function escapeHandler(evt) {
+  if (evt.key === "Escape") {
+    closeModal(currentModal);
+  }
+}
+
+function outsideClickHandler(evt) {
+  if (evt.currentTarget === evt.target) {
+    closeModal(currentModal);
+  }
+}
+
+function openModal(modal) {
+  currentModal = modal;
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", escapeHandler);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", escapeHandler);
+}
+
+const modals = Array.from(document.querySelectorAll(".modal"));
+modals.forEach((modal) => {
+  modal.addEventListener("click", outsideClickHandler);
+
+  const exitBtn = modal.querySelector(".modal__exit-btn");
+  exitBtn.addEventListener("click", () => closeModal(modal));
+});
+
+// Profile Modal
+const profileModal = document.querySelector("#profile-modal");
+const profileForm = document.forms["profile-form"];
+const profileSubmitBtn = profileModal.querySelector(".modal__button");
+const profileInputList = profileModal.querySelectorAll(".modal__input");
+
+const setProfileLoading = setLoadingText(profileSubmitBtn, savingText);
+
+function onProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = getInputValues(profileInputList);
+  setProfileLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions